pages/instructorAssessmentGroups/instructorAssessmentGroups.js

```diff
@@ -0,0 +1,205 @@
+const ERR = require('async-stacktrace');
+const express = require('express');
+const router = express.Router();
+const path = require('path');
+const debug = require('debug')('prairielearn:' + path.basename(__filename, '.js'));
+
+const error = require('@prairielearn/prairielib/error');
+const groupUpdate = require('../../lib/group-update');
+const sqldb = require('@prairielearn/prairielib/sql-db');
+const sqlLoader = require('@prairielearn/prairielib/sql-loader');
+
+const sql = sqlLoader.loadSqlEquiv(__filename);
+function obtainInfo(req, res, next){
+    const params = {assessment_id: res.locals.assessment.id};
+    sqldb.query(sql.config_info, params, function(err, result) {
+        if (ERR(err, next)) return;
+        res.locals.isgroup = true;
+        if (result.rowCount == 0) {
+            res.locals.isgroup = false;
+            res.render(__filename.replace(/\.js$/, '.ejs'), res.locals);
+            return;
+        }
+        res.locals.config_info = result.rows[0];
+        res.locals.config_info.defaultmin = res.locals.config_info.minimum || 2;
+        res.locals.config_info.defaultmax = res.locals.config_info.maximum || 5;
+
+        res.locals.config_info.permission = '';
+        if (res.locals.config_info.student_auth_join) {
+            res.locals.config_info.permission += 'join ';
+        }
+        if (res.locals.config_info.student_auth_create) {
+            res.locals.config_info.permission += 'create ';
+        }
+        if (res.locals.config_info.student_auth_quit) {
+            res.locals.config_info.permission += 'quit ';
+        }
+        const params = {
+            assessment_id: res.locals.assessment.id,
+            course_instance_id: res.locals.config_info.course_instance_id,
+        };
+        sqldb.query(sql.assessment_list, params, function(err, result) {
+            if (ERR(err, next)) return;
+            res.locals.assessment_list_rows = result.rows;
+            sqldb.query(sql.select_groups, params, function(err, result) {
+                if (ERR(err, next)) return;
+                res.locals.groups_rows = result.rows;
+                sqldb.query(sql.not_assigned_users, params, function(err, result) {
+                    if (ERR(err, next)) return;
+                    res.locals.not_assigned_users_rows = result.rows;
+                    debug('render page');
+                    res.render(__filename.replace(/\.js$/, '.ejs'), res.locals);
+                });
+            });
+        });
```

Comment:
None of these queries depend on each other, so there's no need to run them sequentially. Consider moving to promises and then using `Promise.all` so that they can actually run concurrently, or collapse these into fewer queries to avoid the round trips to the DB.